pkg/cloudprovider/events/events.go

```diff
@@ -38,3 +42,13 @@ func NodeClaimFailedToResolveNodeClass(nodeClaim *v1beta1.NodeClaim) events.Even
 		DedupeValues:   []string{string(nodeClaim.UID)},
 	}
 }
+
+func FilteredInstanceTypes(nodeClaim *v1beta1.NodeClaim, resolvedNodeClassITs []string, filteredITs []string) events.Event {
+	return events.Event{
+		InvolvedObject: nodeClaim,
+		Type:           v1.EventTypeNormal,
+		Message: fmt.Sprintf("Resolved instance types from nodeclass: %s, filtered instance types after removing exotic types and spot instance types costlier than cheapest on-demand: %s", utils.PrettySlice(resolvedNodeClassITs, 5),
```

Comment:
```suggestion
		Message: fmt.Sprintf("Filtered instance types after removing exotic types and spot instance types costlier than cheapest on-demand: %s", utils.PrettySlice(resolvedNodeClassITs, 5),
```

I'm not sure that you need the NodeClass bit

Reply:
Even before filtering the exotic types, we filter instanceTypes [here](https://github.com/aws/karpenter-provider-aws/blob/main/pkg/cloudprovider/cloudprovider.go#L247). I believe we would want to have that as part of the event as well?

Reply:
Right, my point was that we don't do that filtering on the "nodeclass", it's done on the "nodeclaim". And the name of the NodeClaim is already included in the event since the involved object here is the NodeClaim